sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -345,10 +345,10 @@ class SQLMetricsSuite extends SparkFunSuite with SQLMetricsTestUtils with Shared
     val df1 = Seq((1, "1"), (2, "2")).toDF("key", "value")
     val df2 = Seq((1, "1"), (2, "2"), (3, "3"), (4, "4")).toDF("key2", "value")
     // Assume the execution plan is
-    // ... -> BroadcastHashJoin(nodeId = 0)
+    // ... -> BroadcastHashJoin(nodeId = 1)
     val df = df1.join(broadcast(df2), $"key" === $"key2", "leftsemi")
     testSparkPlanMetrics(df, 2, Map(
-      0L -> (("BroadcastHashJoin", Map(
+      1L -> (("BroadcastHashJoin", Map(
```

Comment:
why this change? what's changed in the plan?

Reply:
@cloud-fan In this case, we pushdown the leftsemi join below project.
plan before:
```
== Optimized Logical Plan ==
Join LeftSemi, (key#247 = key2#258), rightHint=(broadcast)
:- Project [_1#242 AS key#247, _2#243 AS value#248]
:  +- LocalRelation [_1#242, _2#243]
+- Project [_1#253 AS key2#258]
   +- LocalRelation [_1#253, _2#254]

== Physical Plan ==
*(2) BroadcastHashJoin [key#247], [key2#258], LeftSemi, BuildRight
:- *(2) Project [_1#242 AS key#247, _2#243 AS value#248]
:  +- LocalTableScan [_1#242, _2#243]
+- BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, false] as bigint)))
   +- *(1) Project [_1#253 AS key2#258]
      +- LocalTableScan [_1#253, _2#254]
```
plan after:
```
== Optimized Logical Plan ==
Project [_1#242 AS key#247, _2#243 AS value#248]
+- Join LeftSemi, (_1#242 = key2#258), rightHint=(broadcast)
   :- LocalRelation [_1#242, _2#243]
   +- Project [_1#253 AS key2#258]
      +- LocalRelation [_1#253, _2#254]

== Physical Plan ==
*(2) Project [_1#242 AS key#247, _2#243 AS value#248]
+- *(2) BroadcastHashJoin [_1#242], [key2#258], LeftSemi, BuildRight
   :- LocalTableScan [_1#242, _2#243]
   +- BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, false] as bigint)))
      +- *(1) Project [_1#253 AS key2#258]
         +- LocalTableScan [_1#253, _2#254]
```